test/e2e/apimachinery/namespace.go

```diff
@@ -551,10 +551,26 @@ func ensurePodsAreRemovedFirstInOrderedNamespaceDeletion(ctx context.Context, f
 			if pod.DeletionTimestamp == nil {
 				framework.Failf("Pod %q in namespace %q does not have a metadata.deletionTimestamp set", pod.Name, nsName)
```

Comment:
if we don't see the deletionTimestamp *yet*, we should log and return `false, nil` so we keep polling until the timeout, not instantly fail

Reply:
Updated to retry^^